pkg/querier/querier.go

```diff
@@ -221,12 +204,27 @@ func (qm MergeQuerier) RemoteReadHandler(w http.ResponseWriter, r *http.Request)
 				return
 			}
 
-			matrix, err := qm.Query(ctx, from, to, matchers...)
+			iterators, err := qm.QueryRange(ctx, from, to, matchers...)
 			if err != nil {
 				errors <- err
 				return
 			}
 
+			in := metric.Interval{
+				OldestInclusive: from,
+				NewestInclusive: to,
+			}
+
+			// Convert iterators to matrix
+			matrix := make(model.Matrix, 0, len(iterators))
+			for _, it := range iterators {
+				ss := &model.SampleStream{
+					Metric: it.Metric().Metric,
+					Values: it.RangeValues(in),
```

Comment:
Just to point out: `RangeValues` could return `nil`. Does it matter? What do we do with these `SampleStream`s afterwards?